src/main/java/com/aws/greengrass/clientdevices/auth/ClientDevicesAuthService.java

```diff
@@ -107,6 +110,12 @@ private int getValidCloudCallQueueSize(Topics topics) {
     }
 
     private void initializeInfrastructure() {
+        context.put(RuntimeConfiguration.class, RuntimeConfiguration.from(getRuntimeConfig()));
+
+        // Register IoT repositories
+        context.put(CertificateRepository.class, context.get(CertificateRegistry.class));
```

Comment:
Instead hard-wiring implementations in the context, how about injecting implementations through the constructor, like this:
```
class A {
   private final CertificateRepository certRepo;
   
   @Inject
   public A(CertificateRegistry registry) {
        this.certRepo = registry;
   }
}
```

(Though I don't have a strong feeling about hard-wiring in the context considering CertificateRepository is envisioned to have only one one impl: CertificateRegistry)